src/Lumper.UI/App.axaml.cs

```diff
@@ -17,11 +18,22 @@ public override void OnFrameworkInitializationCompleted()
     {
         if (ApplicationLifetime is IClassicDesktopStyleApplicationLifetime
             desktop)
+        {
+            var vm = new MainWindowViewModel();
             desktop.MainWindow = new MainWindow
             {
-                DataContext = new MainWindowViewModel()
+                DataContext = vm,
             };
 
+            desktop.MainWindow.AddHandler(DragDrop.DragOverEvent, MainWindowViewModel.DragOver);
+            desktop.MainWindow.AddHandler(DragDrop.DropEvent, vm.Drop);
```

Comment:
This can be in MainWindow.axaml.cs because it only adds events to the MainWindow.

Reply:
Correct-- Tl;dr the stuff in the brackets happens AFTER the constructor code. DataContext is not set until afterwards. Hence, I "hid" the DataContext parameter and used a initer/getter to configure things at init.

There is no setter present, but it does get the job done.

Reply:
The brackets are roughly equivalent to

```csharp
var bar = new foo();
bar.x = 5;
bar.y = 6;
```

not

```csharp
bar.x = 5;
bar.y = 6;
var bar = new foo();
```

Reply:
Still creates new objects in the App.axaml.cs though.

Reply:
That's a one-line revert.